numba/np/random/distributions.py

```diff
@@ -529,3 +529,48 @@ def random_poisson(bitgen, lam):
 def random_negative_binomial(bitgen, n, p):
     Y = random_gamma(bitgen, n, (1 - p) / p)
     return random_poisson(bitgen, Y)
+
+
+@register_jitable
+def random_noncentral_chisquare(bitgen, df, nonc):
+    if np.isnan(nonc):
+        return np.nan
+
+    if nonc == 0:
+        return random_chisquare(bitgen, df)
+
+    if 1 < df:
+        Chi2 = random_chisquare(bitgen, df - 1)
+        n = random_standard_normal(bitgen) + np.sqrt(nonc)
+        return Chi2 + n * n
+    else:
+        i = random_poisson(bitgen, nonc / 2.0)
+        return random_chisquare(bitgen, df + 2 * i)
+
+
+@register_jitable
+def random_noncentral_f(bitgen, dfnum, dfden, nonc):
+    t = random_noncentral_chisquare(bitgen, dfnum, nonc) * dfden
+    return t / (random_chisquare(bitgen, dfden) * dfnum)
+
+
+@register_jitable
+def random_logseries(bitgen, p):
+    r = np_log1p(-p)
```

Comment:
If `p==1.0` Numba raises a `ZeroDivisionError` (the `error_model` needs setting to `numpy` in the decorator). 

NumPy just hangs, I suspect this is a bug as it ends up as a call to  `log1p(-p)` i.e. `log1p(-1)`, the result of which is `-HUGE_VAL` use of which in following operations ends up as `-inf`, this results in the `while` loop never exiting unless a call to `next_double` just so happens to hit `1.0` such that `(V >= p)` is true. What do you think?